packages/next/src/build/webpack-config.ts

```diff
@@ -727,10 +727,25 @@ export default async function getBaseWebpackConfig(
 
   const crossOrigin = config.crossOrigin
 
+  // The `serverComponentsExternalPackages` should not conflict with
+  // the `transpilePackages`.
+  if (config.experimental.serverComponentsExternalPackages) {
+    const externalPackageConflicts = config.transpilePackages.filter((pkg) =>
+      config.experimental.serverComponentsExternalPackages?.includes(pkg)
+    )
+    if (externalPackageConflicts.length > 0) {
+      throw new Error(
+        `The packages specified in the 'transpilePackages' conflict with the 'serverComponentsExternalPackages': ${externalPackageConflicts.join(
+          ', '
+        )}`
+      )
+    }
+  }
+
   // For original request, such as `package name`
   const optOutBundlingPackages = EXTERNAL_PACKAGES.concat(
     ...(config.experimental.serverComponentsExternalPackages || [])
-  )
+  ).filter((pkg) => !config.transpilePackages.includes(pkg))
```

Comment:
missing an ?

Reply:
I added it, but `transpilePackages` is declared as non-nullable otherwise?

Reply:
nvm. it's nullable